backend/main.py

```diff
@@ -7,10 +7,20 @@
 from loguru import logger
 from oasst_backend.api.v1.api import api_router
 from oasst_backend.config import settings
+from oasst_backend.exceptions import OasstError
 from starlette.middleware.cors import CORSMiddleware
 
 app = fastapi.FastAPI(title=settings.PROJECT_NAME, openapi_url=f"{settings.API_V1_STR}/openapi.json")
 
+
+@app.exception_handler(OasstError)
+async def oasst_exception_handler(request: fastapi.Request, ex: OasstError):
+    logger.error(f"{request.method} {request.url} failed: {repr(ex)}")
```

Comment:
loguru also has `logger.exception` that gives a nice trace of the exception (in the current context) and annotates all parameters with their values, it's really nice to debug, but also very verbose, maybe worth thinking about

Reply:
Yes, I had exactly that first but removed it again: For OasstErrors it is clear where they come from, e.g. they are usually post translation of generic internal errors. We could think about referencing 'inner exceptions' in the class that could be forwarded to the client when a server debug setting is True.

I was also looking into general handlers for uncaught Exceptions. IMO it would make sense to remove the general `excet Exception` handlers and do the translation into 500 Internal Server Error centralized .. there `logger.exception()` would definitely make sense.